sfge/src/main/java/com/salesforce/cli/CliArgParser.java

```diff
@@ -41,6 +41,48 @@ public CLI_ACTION getCliAction(String... args) {
         }
     }
 
+    public static class CatalogArgParser {
+        private static final int ARG_COUNT = 2;
+        // NOTE: This value must match the one for the RuleType enum declared in Constants.ts.
+        private static final String PATHLESS = "pathless";
+        // NOTE: This value must match the one for the RuleType enum declared in Constants.ts.
+        private static final String DFA = "dfa";
+
+        private List<AbstractRule> selectedRules;
+
+        public CatalogArgParser() {
+            selectedRules = new ArrayList<>();
+        }
+
+        public void parseArgs(String... args) throws RuleUtil.RuleNotFoundException {
```

Comment:
Do we already have tests for this class in MainTest.java? If not, would it make sense to have a few more?

Reply:
Not sure. If not, we can trivially add some.